### PR TITLE
Redis Proxy: Fix tag extraction to work with filter stat_prefix

### PR DESCRIPTION
**Pull request by davinci26**
Working towards #21595. Adds tag extraction for redis proxy filter.

Signed-off-by: Sotiris Nanopoulos <sotiris.nanopoulos@reddit.com>

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Review by ggreenway — CHANGES_REQUESTED** [review]
/wait

File: changelogs/current.yaml [hunk 1]
```diff
@@ -15,6 +15,11 @@ behavior_changes:
     Fixed metric tag extraction so that :ref:`stat_prefix <envoy_v3_api_field_extensions.filters.network.thrift_proxy.v3.ThriftProxy.stat_prefix>`
     is properly extracted. This changes the Prometheus name from
     envoy_thrift_myprefix_request{} to envoy_thrift_request{envoy_thrift_prefix="myprefix"}.
+- area: redis_proxy
```
[the remark this answers is not in the document]
Comment (davinci26) [reply changelogs/current.yaml:18]: If I was keeping up with envoy versions and went through the change-log I would focus more on extensions that I know that I use. So if I use redis proxy and I see the redis proxy area I would read it.

That being said I haven't actually done that my self so I am just speculating here. Should we change it?

[the remark this answers is not in the document]
Comment (davinci26) [reply changelogs/current.yaml:18]: I wonder if we can have something that combines the both like `stats_redis_proxy`?

[the remark this answers is not in the document]
Comment (davinci26) [reply changelogs/current.yaml:18]: In some other case we used `- area: prometheus stats`

**Review by ggreenway — APPROVED** [review]
LGTM

**Pull request merged**